javascript/frameworks/cap/lib/advanced_security/javascript/frameworks/cap/CDS.qll

```diff
@@ -739,5 +849,171 @@ class HandlerParameterData instanceof PropRead {
+
+  CqlQueryRunnerCall() {
+    this = base.getAMemberInvocation(methodName) and
+    (
+      /*
+       * 1. Method call on the CDS facade or the base database service,
+       * accessed as `cds.db`.
+       */
+
+      exists(CdsFacade cds | base = cds.asSource()) or
+      exists(CdsDb cdsDb | base = cdsDb) or
+      /*
+       * 2. Method call on a service instance object.
+       */
+
+      exists(ServiceInstance srv | base.getALocalSource() = srv) or
+      /*
+       * 3. Method call on a transaction object.
+       */
+
+      exists(CdsTransaction tx | base = tx)
+    )
+  }
+
+  /**
+   * Gets an argument to this runner call, including the subsequent builder functions
+   * called in a chained manner on this one.
+   */
+  abstract DataFlow::Node getAQueryParameter();
+}
+
+class CqlRunMethodCall extends CqlQueryRunnerCall {
+  CqlRunMethodCall() { this.getMethodName() = "run" }
+
+  override DataFlow::Node getAQueryParameter() { result = this.getArgument(0) }
+}
+
+/**
+ * A [CRUD-style call](https://cap.cloud.sap/docs/node.js/core-services#crud-style-api)
+ * that translates to running a CQL query internally.
+ */
+class CqlShortcutMethodCall extends CqlQueryRunnerCall {
+  CqlShortcutMethodCall() {
+    this.getMethodName() = ["read", "create", "update", "delete", "insert", "upsert"]
+  }
+
+  abstract override DataFlow::Node getAQueryParameter();
+}
+
+class CqlReadMethodCall extends CqlShortcutMethodCall {
+  CqlReadMethodCall() { this.getMethodName() = "read" }
+
+  override DataFlow::Node getAQueryParameter() {
+    result = this.getAChainedMethodCall(_).getAnArgument()
+  }
+}
+
+class CqlCreateMethodCall extends CqlShortcutMethodCall {
+  CqlCreateMethodCall() { this.getMethodName() = "create" }
+
+  override DataFlow::Node getAQueryParameter() {
+    exists(DataFlow::CallNode chainedMethodCall |
+      chainedMethodCall = this.getAChainedMethodCall(_)
+    |
+      result = chainedMethodCall.getAnArgument() or
+      result = chainedMethodCall.getAnArgument().(SourceNode).getAPropertyWrite().getRhs()
+    )
+  }
+}
+
+class CqlUpdateMethodCall extends CqlShortcutMethodCall {
+  CqlUpdateMethodCall() { this.getMethodName() = "update" }
+
+  override DataFlow::Node getAQueryParameter() {
+    result = this.getAChainedMethodCall(_).getAnArgument()
+  }
+}
+
+class CqlDeleteMethodCall extends CqlShortcutMethodCall {
+  CqlDeleteMethodCall() { this.getMethodName() = "delete" }
+
+  override DataFlow::Node getAQueryParameter() {
+    result = this.getAChainedMethodCall(_).getAnArgument()
+  }
+}
+
+class CqlInsertMethodCall extends CqlShortcutMethodCall {
+  CqlInsertMethodCall() { this.getMethodName() = "insert" }
+
+  override DataFlow::Node getAQueryParameter() {
+    exists(DataFlow::CallNode chainedMethodCall |
+      chainedMethodCall = this.getAChainedMethodCall(_)
+    |
+      result = chainedMethodCall.getAnArgument() or
+      result = chainedMethodCall.getAnArgument().(SourceNode).getAPropertyWrite().getRhs()
```

Comment:
I think we want to remove this case, otherwise it will still be a sink via `StringConcatParameterOfCqlShortcutMethodCall`.

Reply:
Removed the below case in [3b073eb](https://github.com/advanced-security/codeql-sap-js/pull/200/commits/3b073eb47bb12700ca0a24bbdd2e4fcb429ada3b).